projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
-            if difference < tolerance:
-                return True
-            else:  # if one angle is close to 0 and the other close to 4*pi:
-                if difference > 4 * math.pi - tolerance:
-                    return True
-                else:
-                    return False
-        else:
```

Comment:
How about we just remove the last to `else:` and just add here `return False` to simplify the function?